spec/controllers/finders_controller_spec.rb

```diff
@@ -268,6 +268,14 @@
         expect(response).to render_template("finders/show")
       end
 
+      it "correctly renders the new template if the param is given" do
+        get :show, params: {
+          slug: "search/all", keywords: "hello", enable_new_all_content_finder_ui: "true"
```

Comment:
Given that the ENV VAR is checked first, you'll to unset the stubbed constant first, to be sure the param is being tested.

i.e. remove this: 
``` ruby
stub_const("ENV", ENV.to_hash.merge("ENABLE_NEW_SITE_SEARCH_UI" => "true"))
```

Reply:
Ah – this is outside the describe block that stubs the env var to `true`, but that raises a different point in that this set of contexts is hard to read I need to rearrange this whole block a bit better!